### PR TITLE
feat: integrate cex on-ramp cede.store widget

### PR DESCRIPTION
**Pull request by 0xnikter**
## Description
Following our discussion regarding the CEX On-Ramp widget integration with cede.store, I integrate the cede.store widget allowing to transfer funds from centralized exchanges to a wallet connetected to Polkadex.

## Changes Made

Added a widget page
image

Added a sidebar link to on-ramp widget page:
image

Added an on-ramp link to the modal:
image

## How to Test

Check this page: http://localhost:3000/cexOnRamp
You can downoad cede.store extension, add a cex and try to withdraw. If you don't want to import your API keys or don't have a CEX account, you can use the extension in the demo mode to check the withdrawal.

## Checklist

- [ ] Included link to corresponding [Polkadex Orderbook Frontend Issue](https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/issues).
- [x] I have tested these changes thoroughly.
- [ ] I have requested a review from at least one other contributor.

### CODE REVIEW

**Comment by nuel77** [comment]
hey @neeeekitos thanks for the contirbution. we will check this and added it along with the next weekly release.

File: apps/hestia/src/components/cexOnRamp/template.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import { renderSendWidget } from "@cedelabs/widgets-universal";
+import { useConnectWalletProvider } from "@orderbook/core/providers/user/connectWalletProvider";
+import { Typography } from "@polkadex/ux";
+import { RiInformation2Line } from "@remixicon/react";
+import { Fragment, useEffect, useMemo, useRef } from "react";
+import { useResizeObserver, useWindowSize } from "usehooks-ts";
+
+import { Help } from "../transfer/Help";
```
Comment (ap211unitech) [inline apps/hestia/src/components/cexOnRamp/template.tsx]: We are using component based project structure so all the components required for a page, should be in current folder only. 

For e.g. You can't import `Help` component from transfer page. I would suggest you to create a Help component in current directory.

File: apps/hestia/src/components/cexOnRamp/template.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { SelectAsset } from "../transfer/SelectAsset";
```
Comment (ap211unitech) [inline apps/hestia/src/components/cexOnRamp/template.tsx]: Same here

File: apps/hestia/src/app/cexOnRamp/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { ReactNode } from "react";
+
+const WithdrawsProvider = dynamic(
+  () =>
+    import("@orderbook/core/providers").then((mod) => mod.WithdrawsProvider),
+  { ssr: false }
+);
+
+export default function Layout({ children }: { children: ReactNode }) {
+  return <WithdrawsProvider>{children}</WithdrawsProvider>;
```
Comment (ap211unitech) [inline apps/hestia/src/app/cexOnRamp/layout.tsx]: I don't think `WithdrawProvider` is required here. Can you remove it ?

Comment (ap211unitech) [reply apps/hestia/src/app/cexOnRamp/layout.tsx]: You can rather delete this file, there would be no different layout required here. We already have a layout file in `app` folder.

File: apps/hestia/src/components/cexOnRamp/template.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    useConnectWalletProvider();
+
+  useEffect(() => {
+    renderSendWidget("#cede-widget", {
+      theme: {
+        borderRadius: -15,
+        borderColor: themeConfig.theme.extend.colors["secondary-base"],
+        inputBorderColor: themeConfig.theme.extend.colors["secondary-base"],
+        logoBorderColor: themeConfig.theme.extend.colors["primary-base"],
+        accentColor: themeConfig.theme.extend.colors["primary-base"],
```
Comment (ap211unitech) [inline apps/hestia/src/components/cexOnRamp/template.tsx]: Please refactor it a bit with a variable. 

```js
const extend = themeConfig.theme.extend; // Define it outside the component.
```

and use it like this.

```js
borderColor: extend.colors["secondary-base"]
```

Check [this](https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/blob/release/apps/hestia/src/components/trading/Graph/config.ts#L11) out for help.

**Review by ap211unitech — CHANGES_REQUESTED** [review]

**Comment by 0xnikter** [comment]
Hey @ap211unitech, thank you for your review. I've implemented the required fixes and cleaned up the `cexOnRamp` page

File: apps/hestia/src/components/cexOnRamp/template.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        logoBorderColor: extend.colors["primary-base"],
+        accentColor: extend.colors["primary-base"],
+      },
+      config: {
+        tokenSymbol: "PDEX",
+        network: "polkadex",
+        address: selectedWallet?.address,
+      },
+    });
+  }, []);
```
Comment (ap211unitech) [inline apps/hestia/src/components/cexOnRamp/template.tsx]: Missing dependency values here.

File: apps/hestia/src/components/cexOnRamp/template.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    box: "border-box",
+  });
+
+  const { selectedWallet, browserAccountPresent, extensionAccountPresent } =
+    useConnectWalletProvider();
+
+  useEffect(() => {
+    renderSendWidget("#cede-widget", {
+      theme: {
+        borderRadius: -15,
```
Comment (ap211unitech) [inline apps/hestia/src/components/cexOnRamp/template.tsx]: Border radius in negative ? Doesn't make sense.

Comment (ap211unitech) [reply apps/hestia/src/components/cexOnRamp/template.tsx]: Put 1 instead

**Review by ap211unitech — CHANGES_REQUESTED** [review]
image

The  `#cede.widget` component doesn't look properly aligned. It's not properly vertically aligned.

**Comment by 0xnikter** [comment]
@ap211unitech done ✅

**Review by ap211unitech — APPROVED** [review]

**Comment by ap211unitech** [comment]
@neeeekitos Great. Approved it.

**Review by minikas — APPROVED** [review]

**Pull request merged**